test/unit/lib/system/deploy_test.rb

```diff
@@ -15,32 +15,85 @@ class DeployTest < ActiveSupport::TestCase
     assert_equal 'x', System::Deploy.info.minor_version
   end
 
-  class OnpremisesDeployTest < DeployTest
+  test 'parse deploy info' do
+    File.open('.deploy_info_test', 'w') do |file|
+      file << '{"revision": "rev", "release": "rel"}'
+      file.flush
+
+      deploy_info = System::Deploy.parse_deploy_info(file.path)
+      assert_equal 'rev', deploy_info['revision']
+      assert_equal 'rel', deploy_info['release']
+    end
+  end
+
+  test 'call' do
+    System::Deploy.load_info!
+    response = System::Deploy.call(nil)
+    assert_equal 200, response[0]
+    assert_equal 'application/json', response[1]['Content-Type']
+    assert_equal '2.x', JSON.parse(response[2].first)['release']
+  end
+
+  class OnpremisesDeployTest < ActiveSupport::TestCase
     setup do
       ThreeScale.config.stubs(onpremises: true)
+      System::Deploy.load_info! ActiveSupport::JSON.decode('{"revision": "2.13-stable", "release": "2.13"}')
```

Comment:
should we revert to original state in `teardown`?

Reply:
I'm not sure it's needed, as the `setup` will kind of set it up again from scratch.
But well, I will add it just in case.

Reply:
I mean that other tests might be affected by current (non-)saas setting.

Reply:
OK, fair enough